### ISSUE TITLE
Contribute a bug fix

### ISSUE THREAD

**Issue by JVimes** [dnGrep/dnGrep#215]
_From [estarb...@gmail.com](https://code.google.com/u/104520110858125106359/) on July 31, 2014 08:33:24_

I would like to be able to contribute a bug fix.  I have a clone of the repository.  The bug fix is at https://code.google.com/r/estarback-dngrep/source/detail?r=f1c4b2205b9a864f937911c673945e7751b2a3e2 The issues is that when you have "Preview file" selected, and the preview is open.  If you selection Options and close the Options.  Then when you select an item that would show up in the Preview window it throws an exception.

_Original issue: http://code.google.com/p/dngrep/issues/detail?id=212_

**Comment by doug24** [comment]
This has been fixed in the 2.9.21 beta.

**Comment by JVimes** [comment]
Crash is fixed. However, after closing the options window, preview doesn't reopen. And clicking on the same file does not open it (have to click on a different file). @doug24, is this an easy fix, or should I open a new issue? Thanks again for all the development!

**Comment by doug24** [comment]
Doesn't look like an easy fix.  The preview window is opened on a selection change event in the results. When it closed, the selection doesn't reset, so when clicking it again there is no selection change.  I wonder why the preview window is closed when either the Options window or the Test window is opened?  But changing that will take a bit to unwind, as it's tied into saving the user settings.

**Comment by JVimes** [comment]
Good question. When Options or Test are closed, we could reopen the Preview window contingent on `PreviewFileContent`. That's letting the hack grow. Which is okay as a stop-gap, but I logged #250, so we can fix it the correct way eventually. I'll close this issue, but I'm okay with a stop-gap.
